hawp/detector.py

```diff
@@ -120,14 +120,18 @@ def forward(self, images, annotations = None):
         #jloc_pred_nms = non_maximum_suppression(jloc_pred[0])
         #topK = torch.clamp((jloc_pred_nms > 0.008).count_nonzero(), max=300)
 
-        juncs_pred, _ = get_junctions(non_maximum_suppression(jloc_pred[0]),joff_pred[0], topk=300, th=0.008)
+        nms_jloc_pred = non_maximum_suppression(jloc_pred)[0]
+        juncs_pred, _ = get_junctions(nms_jloc_pred,joff_pred[0], topk=300, th=0.008)
         extra_info['time_proposal'] = time.time() - extra_info['time_proposal']
         extra_info['time_matching'] = time.time()
         dis_junc_to_end1, idx_junc_to_end1 = torch.sum((lines_pred[:,:2]-juncs_pred[:,None])**2,dim=-1).min(0)
         dis_junc_to_end2, idx_junc_to_end2 = torch.sum((lines_pred[:,2:] - juncs_pred[:, None]) ** 2, dim=-1).min(0)
 
-        idx_junc_to_end_min = torch.min(idx_junc_to_end1,idx_junc_to_end2)
-        idx_junc_to_end_max = torch.max(idx_junc_to_end1,idx_junc_to_end2)
+        # idx_junc_to_end_min = torch.min(idx_junc_to_end1,idx_junc_to_end2)
+        # idx_junc_to_end_max = torch.max(idx_junc_to_end1,idx_junc_to_end2)
```

Comment:
torch.min and torch.max signatures are (tensor, dim, keep_dims, ..)
They are used here are torch.min/max(tensor1, tensor2) to do element-wise minimum/maximum operations.
This does not work for onnx, as it sees min and max, and expects the second argument to be an integer, and not a tensor.

The fix is to stack the two tensors together, then take the min/max along the stacked dimension.